format-all.el

```diff
@@ -1355,6 +1355,18 @@ Consult the existing formatters for examples of BODY."
                                   (line-number-at-pos (car region))
                                   (line-number-at-pos (cdr region))))))))
 
+(define-format-all-formatter syntax_tree
+  (:executable "stree")
+  (:install "gem install syntax_tree:'>=2.0.1'")
+  (:languages "Ruby")
+  (:features)
+  (:format
+   (format-all--buffer-hard-ruby
+    "syntax_tree" '(0 1) nil '(".streerc")
+    executable
+    "format"
+    (or (buffer-file-name) (buffer-name)))))
```

Comment:
Does this tell stree to format from stdin to stdout?

Reply:
This will not work reliably. According to [the readme](https://github.com/ruby-syntax-tree/syntax_tree#format):

> This command will output the formatted version of each of the listed files. Importantly, it will not write that content back to the source files. It is meant to display the formatted version only.
>
> `stree format path/to/file.rb`

It's good that stree writes the formatted code to stdout, but format-all also needs to read the unformatted code from stdin. Reading from a file is not reliable, since the Emacs user may not have saved the file.

Reply:
Sorry I had not gotten back to this yet. You are correct. The good news is:
> For all of these commands, you can alternatively pass in content through STDIN

So we should be good to go by dropping the filename. I will spend some time testing it again

Reply:
Tests were successful with the latest changes. 
1. Explicitly invoking the formatted via `format-all-buffer`
2. Formatting on save via `format-all-mode`